src/main/java/mate/academy/store/service/order/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,125 @@
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public ResponseOrderDto placeNewOrder(Long userId, RequestOrderDto requestOrderDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find users shopping cart by user id: "
+                    + userId));
+
+        Order order = new Order();
+        Set<OrderItem> orderItems = createSetOfOrderItems(order, shoppingCart);
+
+        order.setOrderItems(orderItems);
+        order.setTotal(countTotal(orderItems));
+        order.setShippingAddress(requestOrderDto.getShippingAddress());
+        order.setUser(shoppingCart.getUser());
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Status.PENDING);
+
+        shoppingCart.clearCart();
+        shoppingCartRepository.save(shoppingCart);
+
+        Order savedOrder = orderRepository.save(order);
+
+        orderItems.forEach(this::saveOrderItem);
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    public Set<ResponseOrderDto> getAllUsersOrders(Long userId) {
+        return orderRepository.findAllByUserId(userId)
+            .stream()
+            .map(orderMapper::toDto)
+            .collect(Collectors.toSet());
+    }
+
+    @Override
+    @Transactional
+    public ResponseOrderDto updateOrderStatus(Long id, UpdateOrderRequestDto requestDto) {
+        Order order = orderRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order by id: " + id));
+        orderMapper.updateOrderFromDto(requestDto, order);
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public OrderItemDto getItemByOrderIdAndItemId(Long orderId, Long id) {
+        OrderItem orderItem = orderItemRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find item by id: " + id));
+
+        if (orderItem.getOrder().getId().equals(orderId)) {
+            return orderItemMapper.toDto(orderItem);
+        }
+        throw new EntityNotFoundException("Can't find item by order id: " + orderId);
+    }
+
+    @Override
+    public Set<OrderItemDto> getItemsByOrderId(Long id) {
+        return orderItemRepository.findAllByOrderId(id)
+            .stream()
+            .map(orderItemMapper::toDto)
+            .collect(Collectors.toSet());
+    }
+
+    @Transactional
+    private void saveOrderItem(OrderItem orderItem) {
+        orderItemRepository.save(orderItem);
+    }
```

Comment:
Why do you need this? transaction in transaction)))

Reply:
> Why do you need this? transaction in transaction)))

So, if the method that called saveOrderItem already has a @Transactional annotation over it, then i no longer need to use it over saveOrderItem? Is this what you meant? I just thought that this annotation should be used in all methods that create or change information in the DB